mlflow/models/utils.py

```diff
@@ -208,15 +240,24 @@ def _handle_dataframe_input(df):
                     f"but got '{type(input_example)}'",
                 )
             self.data = _handle_dataframe_input(self._inference_data)
-            self.info = {
-                "artifact_path": example_filename,
-                "type": "dataframe",
-                "pandas_orient": "split",
-            }
+            self.info.update(
+                {
+                    "type": "dataframe",
+                    "pandas_orient": "split",
+                }
+            )
 
     def save(self, parent_dir_path: str):
         """Save the example as json at ``parent_dir_path``/`self.info['artifact_path']`."""
-        with open(os.path.join(parent_dir_path, self.info["artifact_path"]), "w") as f:
+        if self._inference_params is not None:
+            if "params" in self.data:
+                raise MlflowException.invalid_parameter_value(
+                    "The input example contains 'params' key, which is reserved for "
```

Comment:
We should use interpolation for this in the exception message so that the constant's reserved name can be dynamically provided in the message.